src/tests-tokens/tokens-metadata-tests.ts

```diff
@@ -0,0 +1,280 @@
+      `/extended/v1/tokens/${senderAddress}.beeple/nft/metadata`
+    );
+    expect(query1.status).toBe(200);
+    expect(query1.body).toHaveProperty('token_uri');
+    expect(query1.body).toHaveProperty('name');
+    expect(query1.body).toHaveProperty('description');
+    expect(query1.body).toHaveProperty('image_uri');
+    expect(query1.body).toHaveProperty('image_canonical_uri');
+  });
+
+  test('token nft-metadata data URL base64 w/o media type', async () => {
+    const contract1 = await deployContract(
+      'beeple',
+      pKey,
+      'src/tests-tokens/test-contracts/beeple-data-url-b.clar',
+      api
+    );
+    const tx1 = await standByForTx(contract1.txId);
+    if (tx1.status != 1) logger.error('contract deploy error', tx1);
+
+    await standByForTokens(contract1.contractId);
+
+    const query = await db.getNftMetadata(contract1.contractId);
+    expect(query.found).toBe(true);
+
+    const senderAddress = getAddressFromPrivateKey(pKey, stacksNetwork.version);
+    const query1 = await supertest(api.server).get(
+      `/extended/v1/tokens/${senderAddress}.beeple/nft/metadata`
+    );
+    expect(query1.status).toBe(200);
+    expect(query1.body).toHaveProperty('token_uri');
+    expect(query1.body).toHaveProperty('name');
+    expect(query1.body).toHaveProperty('description');
+    expect(query1.body).toHaveProperty('image_uri');
+    expect(query1.body).toHaveProperty('image_canonical_uri');
+  });
+
+  test('token nft-metadata data URL plain non-encoded', async () => {
+    const contract1 = await deployContract(
+      'beeple',
+      pKey,
+      'src/tests-tokens/test-contracts/beeple-data-url-c.clar',
+      api
+    );
+    const tx1 = await standByForTx(contract1.txId);
+    if (tx1.status != 1) logger.error('contract deploy error', tx1);
+
+    await standByForTokens(contract1.contractId);
+
+    const query = await db.getNftMetadata(contract1.contractId);
+    expect(query.found).toBe(true);
+
+    const senderAddress = getAddressFromPrivateKey(pKey, stacksNetwork.version);
+    const query1 = await supertest(api.server).get(
+      `/extended/v1/tokens/${senderAddress}.beeple/nft/metadata`
+    );
+    expect(query1.status).toBe(200);
+    expect(query1.body).toHaveProperty('token_uri');
+    expect(query1.body).toHaveProperty('name');
+    expect(query1.body).toHaveProperty('description');
+    expect(query1.body).toHaveProperty('image_uri');
+    expect(query1.body).toHaveProperty('image_canonical_uri');
+  });
+
+  test('token nft-metadata', async () => {
+    const contract = await deployContract(
+      'nft-trait',
+      pKey,
+      'src/tests-tokens/test-contracts/nft-trait.clar',
+      api
+    );
+    const tx = await standByForTx(contract.txId);
+    if (tx.status != 1) logger.error('contract deploy error', tx);
+
+    // TODO: these example contracts need the 3rd party `get-token-uri` urls fetch-mocked
+    // so these tests pass if/when those resources become unavailable or change.
```

Comment:
@asimm241 I added a comment here regarding unit test dependencies on 3rd party resources. These tests can break at any moment, we need to implement fetch mocking to make these durable.